scipy/stats/tests/test_stats.py

```diff
@@ -1875,65 +1875,99 @@ def test_zscore_nan_raise(self):
         assert_equal(mad_axis0, np.nan)
-        mad_axis1 = stats.median_absolute_deviation(z, axis=1)
         assert_equal(mad_axis1, np.array([np.nan, np.nan, np.nan]))
         assert_equal(mad_axis1.shape, (3,))
 
     def test_mad_nan_shape2(self):
         z = np.ones((3, 0, 2))
-        mad_axis0 = stats.median_absolute_deviation(z, axis=0)
+        with suppress_warnings() as sup:
+            sup.filter(DeprecationWarning)
+            mad_axis0 = stats.median_absolute_deviation(z, axis=0)
+            mad_axis1 = stats.median_absolute_deviation(z, axis=1)
+            mad_axis2 = stats.median_absolute_deviation(z, axis=2)
         assert_equal(mad_axis0, np.nan)
-        mad_axis1 = stats.median_absolute_deviation(z, axis=1)
         assert_equal(mad_axis1, np.array([[np.nan, np.nan],
                                           [np.nan, np.nan],
                                           [np.nan, np.nan]]))
         assert_equal(mad_axis1.shape, (3, 2))
-        mad_axis2 = stats.median_absolute_deviation(z, axis=2)
         assert_equal(mad_axis2, np.nan)
 
     def test_mad_nan_propagate(self):
-        dat = np.array([2.20, 2.20, 2.4, 2.4, 2.5, 2.7, 2.8, 2.9, 3.03,
-                3.03, 3.10, 3.37, 3.4, 3.4, 3.4, 3.5, 3.6, 3.7, 3.7,
-                3.7, 3.7,3.77, 5.28, np.nan])
-
-        mad = stats.median_absolute_deviation(dat, nan_policy='propagate')
+        with suppress_warnings() as sup:
+            sup.filter(DeprecationWarning)
+            mad = stats.median_absolute_deviation(self.dat_nan,
+                                                  nan_policy='propagate')
         assert_equal(mad, np.nan)
 
     def test_mad_nan_raise(self):
-        dat = np.array([2.20, 2.20, 2.4, 2.4, 2.5, 2.7, 2.8, 2.9, 3.03,
-                3.03, 3.10, 3.37, 3.4, 3.4, 3.4, 3.5, 3.6, 3.7, 3.7,
-                3.7, 3.7,3.77, 5.28, np.nan])
-
         with assert_raises(ValueError):
-            stats.median_absolute_deviation(dat, nan_policy='raise')
+            with suppress_warnings() as sup:
+                sup.filter(DeprecationWarning)
+                stats.median_absolute_deviation(self.dat_nan,
+                                                nan_policy='raise')
 
-    def test_mad_nan_omit(self):
-        dat = np.array([2.20, 2.20, 2.4, 2.4, 2.5, 2.7, 2.8, 2.9, 3.03,
-                3.03, 3.10, 3.37, 3.4, 3.4, 3.4, 3.5, 3.6, 3.7, 3.7,
-                3.7, 3.7,3.77, 5.28, np.nan])
+    def test_mad_scale_default(self):
+        with suppress_warnings() as sup:
+            sup.filter(DeprecationWarning)
+            mad = stats.median_absolute_deviation(self.dat, scale="raw")
+            mad_float = stats.median_absolute_deviation(self.dat, scale=1.0)
+        assert_almost_equal(mad, 0.355)
+        assert_almost_equal(mad, mad_float)
 
-        mad = stats.median_absolute_deviation(dat, nan_policy='omit')
-        assert_almost_equal(mad, 0.504084)
+    def test_mad_scale_default(self):
```

Comment:
Rename this test; this name is already used in line 1955.  (`flake8` caught this.)

Reply:
renamed to `test_mad_scale_normal` and rebased the branch in latest (force) push

Reply:
+1 for rebasing and squashing the commits.  Doing that is the perfect time to also clean up the commit message.  We no longer need that itemized list of commit messages for individual commits that no longer exist.  Can you simplify the commit message so that it only explains what actually changed in the final version of the PR?

Reply:
I'll update the commit message when I merge this.  The docstrings need a couple more small tweaks, which I'm working on now.  Unfortunately, I'm still not able to build the docs locally, so I'll have to check the result generated by CircleCI.